app/scripts/metamask-controller.js

```diff
@@ -4643,34 +4654,6 @@ export default class MetamaskController extends EventEmitter {
     }
   }
 
-  /**
-   * Encodes a BIP-39 mnemonic as the indices of words in the English BIP-39 wordlist.
-   *
-   * @param {Buffer} mnemonic - The BIP-39 mnemonic.
-   * @returns {Buffer} The Unicode code points for the seed phrase formed from the words in the wordlist.
-   */
-  _convertMnemonicToWordlistIndices(mnemonic) {
-    const indices = mnemonic
-      .toString()
-      .split(' ')
-      .map((word) => wordlist.indexOf(word));
```

Comment:
This would return `-1` if the word didn't exist! This causes `Uint16Array` to hold the value `[65535]`, which results in a Uint8Array of `[255, 255]`. Which is invalid. The new version of this function in mnemonic.ts throws when it receives an invalid word.